superbench/common/utils/network.py

```diff
@@ -0,0 +1,41 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Network Utility."""
+
+import socket
+import subprocess
+from contextlib import closing
+
+
+def get_free_port():
+    """Get a free port in current system.
+
+    Return:
+        port (int): a free port in current system.
+    """
+    with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as s:
+        s.bind(('', 0))
+        s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        return s.getsockname()[1]
+
+
+def get_ib_devices():
+    """Get available IB devices in the system and filter ethernet devices.
+
+    Return:
+        ib_devices (list): IB devices in current system.
+    """
+    # Filter 'InfiniBand' devices by link_layer
+    command = "ibv_devinfo | awk '$1 ~ /hca_id/||/link_layer:/ {print $1,$2}'"
```

Comment:
check `/sys/class/infiniband/<device>/ports/<port-num>/link_layer` is `InifiniBand` and get `<device>` list?

reference: [sysfs-class-infiniband](https://www.kernel.org/doc/Documentation/ABI/stable/sysfs-class-infiniband)

Reply:
could also check `/sys/class/infiniband/<device>/node_type` for transport type
referemce: [dev->transport_type](https://github.com/linux-rdma/rdma-core/blob/d7a9dde2b164a1563c7a51e4ade8ea3b6cdea9ef/libibverbs/init.c#L380-L401)

Reply:
thank you very much